joblib/numpy_pickle.py

```diff
@@ -6,32 +6,35 @@
 # Copyright (c) 2009 Gael Varoquaux
 # License: BSD Style, 3 clauses.
 
+import os
 import pickle
-import traceback
 import sys
-import os
-import zlib
+import traceback
 import warnings
-import struct
-import codecs
+import zlib
+from io import BytesIO
 
 from ._compat import _basestring
-
-from io import BytesIO
+from .externals import dill
 
 PY3 = sys.version_info[0] >= 3
+Pickler = dill.Pickler
+Unpickler = dill.Unpickler
 
-if PY3:
-    Unpickler = pickle._Unpickler
-    Pickler = pickle._Pickler
 
+def set_pickle(Pickler_, Unpickler_):
```

Comment:
Does that mean that backwards compatibility is already tested, since dill is now default and the pickled files are generated using previous versions without dill.